### PR TITLE
refactor(case/reports): automatically resolve pending reports on case creation

### LINKED ISSUE #1138: Resolve report if an action is taken on the report target
Currently, we require to explicitly set case references during or after moderation actions.

Suggestion: Resolve all pending cases, if a moderation action is taken (on author or target, as defined in manual setting atm)
If something goes horribly wrong, we always have the option to manually set state/reference, if needed.

### PR DESCRIPTION
**Pull request by JPBM135**
## Why? 
Closes #1138 and a proper explanation is present on that issue

> **Note**
> This PR requires a command re-deploy (descriptions updated)

### CODE REVIEW

**Comment by JPBM135** [comment]
Need to improve the clarrification based on the approach

**Comment by iCrawl** [comment]
- https://github.com/Naval-Base/yuudachi/pull/1139 merged 👍

**Comment by JPBM135** [comment]
Will perform a refactor and then it will be ready

File: packages/yuudachi/src/functions/reports/resolveReports.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+export async function resolvePendingReports(guild: Guild, targetId: string, caseId: number, moderator: User) {
+	const sql = container.resolve<Sql<any>>(kSQL);
+
+	const pendingReports = await sql<RawReport[]>`
+		select *
+		from reports
+		where guild_id = ${guild.id}
+			and status = ${ReportStatus.Pending}
+			and (target_id = ${targetId} or author_id = ${targetId})
```
Comment (JPBM135) [inline packages/yuudachi/src/functions/reports/resolveReports.ts:18]: Should we resolve reports (as Spam) where the actioned user is the author? Or only the one where it is the target?

Comment (iCrawl) [reply packages/yuudachi/src/functions/reports/resolveReports.ts:18]: I don't think so? The actioned user could make a legitimate report, but still be reported/actioned themselves, no?

Comment (JPBM135) [reply packages/yuudachi/src/functions/reports/resolveReports.ts:18]: Yeah, thats what I thought too. Give me a sec to remove

**Pull request merged**